contents/tutorials/anthropic-analytics.md

```diff
@@ -0,0 +1,357 @@
+
+```bash
+npx create-next-app@latest anthropic-analytics
+cd anthropic-analytics
+npm install --save @anthropic-ai/sdk
+npm install --save posthog-js
+cd ./src/app
+touch providers.js # we set up PostHog in this file below
+```
+
+When prompted, select **No** for TypeScript, **Yes** for `use app router`, **No** for Tailwind CSS and the defaults for every other option.
+
+Next, we set up Posthog using our API key and host (You can find these in [your project settings](https://us.posthog.com/settings/project)). Add the below code to `app/providers.js`:
+
+```js file=app/providers.js
+'use client'
+import posthog from 'posthog-js'
+import { PostHogProvider } from 'posthog-js/react'
+
+if (typeof window !== 'undefined') {
+  posthog.init('<ph_project_api_key>', {
+    api_host: '<ph_client_api_host>',
+  })
+}
+
+export function PHProvider({ children }) {
+  return <PostHogProvider client={posthog}>{children}</PostHogProvider>
+}
+```
+
+Then we import the `PHProvider` component into `app/layout.js` and wrap our app with it:
+
+```js file=app/layout.js
+import { Inter } from "next/font/google";
+import "./globals.css";
+import { PHProvider } from './providers'
+
+const inter = Inter({ subsets: ["latin"] });
+
+export const metadata = {
+  title: "Create Next App",
+  description: "Generated by create next app",
+};
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <PHProvider>
+        <body className={inter.className}>{children}</body>
+      </PHProvider>
+    </html>
+  );
+}
+```
+
+Then replace the code in `page.js` with our basic layout and functionality. You can find your Anthropic API key [here](https://console.anthropic.com/settings/keys).
+
+```js file=app/page.js
+'use client'
+
+import { useState } from 'react';
+import { usePostHog } from 'posthog-js/react'
+import Anthropic from '@anthropic-ai/sdk';
+
+const models = [
+  {
+    name: 'claude-3-opus-20240229',
+    token_input_cost: 0.00001500,
+    token_output_cost: 0.00007500
+  },
+  {
+    name: 'claude-3-sonnet-20240229',
+    token_input_cost: 0.00000300,
+    token_output_cost:  0.00001500
+  },
+    {
+    name: 'claude-3-haiku-20240307',
+    token_input_cost: 0.00000025,
+    token_output_cost: 0.00000125
+  },
+]
+
+export default function Home() {
+  const [userInput, setUserInput] = useState('');
+  const [response, setResponse] = useState('');
+  const [selectedModel, setSelectedModel] = useState(models[0]);
+  const posthog = usePostHog()
+
+  const fetchResponse = async () => {
+    try {
+      const anthropic = new Anthropic({
+        apiKey: '<your_anthropic_api_key>', 
+        baseURL: window.location.origin + '/anthropic/',
```

Comment:
Why can't we just use the API directly?

Reply:
Was getting a cors error: https://www.reddit.com/r/ClaudeAI/comments/1b9pn20/cors_error_in_anthropic_api_while_calling_from/